pkg/storage/swift/swift.go

```diff
@@ -50,35 +139,71 @@ func (d *driver) ConfigEnv() (envs []corev1.EnvVar, err error) {
 				},
 			},
 		},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_REGION", Value: d.Config.RegionName},
 	)
 
-	if d.Config.Domain != "" {
-		envs = append(envs, corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_DOMAIN", Value: d.Config.Domain})
-	}
-
-	if d.Config.DomainID != "" {
-		envs = append(envs, corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_DOMAINID", Value: d.Config.DomainID})
-	}
+	return
+}
 
-	if d.Config.Tenant != "" {
-		envs = append(envs, corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_TENANT", Value: d.Config.Tenant})
+func (d *driver) StorageExists(cr *imageregistryv1.Config) (bool, error) {
+	client, err := d.getSwiftClient(cr)
+	if err != nil {
+		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapi.ConditionUnknown, "Could not connect to registry storage", err.Error())
+		return false, err
 	}
 
-	if d.Config.TenantID != "" {
-		envs = append(envs, corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_TENANTID", Value: d.Config.TenantID})
+	_, err = containers.Get(client, cr.Spec.Storage.Swift.Container, containers.GetOpts{}).Extract()
+	if err != nil {
+		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapi.ConditionFalse, "Storage does not exist", err.Error())
+		return false, err
 	}
-	return
-}
 
-func (d *driver) StorageExists(cr *imageregistryv1.Config) (bool, error) {
-	return false, nil
+	util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapi.ConditionTrue, "Swift container Exists", "")
+	return true, nil
 }
 
 func (d *driver) StorageChanged(cr *imageregistryv1.Config) bool {
+	if !reflect.DeepEqual(cr.Status.Storage.Swift, cr.Spec.Storage.Swift) {
+		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapi.ConditionUnknown, "Swift Configuration Changed", "Swift storage is in an unknown state")
+		return true
+	}
+
 	return false
 }
 
 func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
+	client, err := d.getSwiftClient(cr)
+	if err != nil {
+		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapi.ConditionFalse, err.Error(), err.Error())
+		return err
+	}
+
+	// Generate new container name if it wasn't provided.
+	// The name has a prefix "image_registry_", which is complemented by 16 capital latin letters
+	// Example of a generated name: image_registry_FHEIBGDDGBLWPXFR
+	if cr.Spec.Storage.Swift.Container == "" {
+		bytes := make([]byte, 16)
+		for i := 0; i < 16; i++ {
+			bytes[i] = byte(65 + rand.Intn(25)) // A=65 and Z=65+25
+		}
+		cr.Spec.Storage.Swift.Container = "image_registry_" + string(bytes)
+	}
+
+	fmt.Println(cr.Spec.Storage.Swift.Container)
```

Comment:
Use glog for log messages (github.com/golang/glog) - for debug messages our convention is to use a log level of 5 or higher:
`glog.V(5).Infof("Attempting to create swift container %s", cr.Spec.Storage.Swift.Container)`

Reply:
I decided to completely remove this line - I used it for local debugging and forgot to remove.